### PR TITLE
Resolves players escaping jail early

### LINKED ISSUE #7771: Player can leave jail after relogin
### rAthena Hash

ccd7e5c14a432469b68f0bd731712887deb1224e

### Client Date

20220406

### Server Mode

Renewal

### Result

Player can leave jail after relogin

### Relevant Log Output

_No response_

### Expected Result

Players must not be able to leave jail by relogin

### How to Reproduce

1. gm type @jailfor 5mn Playername
2. Player relogin
3. After this player will leave jail and return last savepoint

### Official Information

.

### Modifications that may affect results

_No response_

### PR DESCRIPTION
**Pull request by aleos89**
* **Addressed Issue(s)**: Fixes #7771

* **Server Mode**: Pre-renewal and Renewal

* **Description of Pull Request**: 
  * Adds a missing check to avoid warping players who are actively jailed out early after relogging.
Thanks to @fab10Carvalho!

### CODE REVIEW

File: src/map/pc.cpp [hunk 1]
```diff
@@ -1926,6 +1926,11 @@ bool pc_lastpoint_special( map_session_data& sd ){
 		return true;
 	}
 
+	if (strcmpi(sd.status.last_point.map, MAP_JAIL) == 0) {
```
Comment (Lemongrass3110) [inline src/map/pc.cpp:1929]: Doesnt that make normal "jail visitors" stuck on that map too?

File: src/map/status.cpp [hunk 2]
```diff
@@ -13415,6 +13415,9 @@ int status_change_end(struct block_list* bl, enum sc_type type, int tid)
 			}
 			break;
 		case SC_JAILED:
+			if (tid == INVALID_TIMER)
```
Comment (Lemongrass3110) [inline src/map/status.cpp:13418]: This will break `@unjail`.

**Review by Lemongrass3110 — CHANGES_REQUESTED** [review]

File: src/map/pc.cpp (hunk 1 above)
Comment (aleos89) [reply src/map/pc.cpp:1929]: It shouldn't. Jail has the nosave mapflag so the player shouldn't have their lastpoint set to the jail map.

Comment (vstumpf) [reply src/map/pc.cpp:1929]: Unfortunately, I don't think there is another way

**Review by vstumpf — APPROVED** [review]

**Review by Lemongrass3110 — APPROVED** [review]

**Pull request merged**